src/runtime/virtcontainers/network_linux.go

```diff
@@ -211,13 +211,18 @@ func (n *LinuxNetwork) addSingleEndpoint(ctx context.Context, s *Sandbox, netInf
 	return endpoint, nil
 }
 
-func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, idx int, hotplug bool) error {
+func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, endpoint Endpoint, hotplug bool) error {
+	var idx int = len(n.eps)
+	for i, val := range n.eps {
+		if val == endpoint {
+			idx = i
+			break
+		}
+	}
 	if idx > len(n.eps)-1 {
 		return fmt.Errorf("Endpoint index overflow")
 	}
```

Comment:
To make it clear, I tend to not to use `idx > len(n.eps)-1`.

```suggestion
	if idx == len(n.eps) {
		return fmt.Errorf("Endpoint not found")
	}
```

Reply:
Thanks for your feedback! I agree with your suggestion to modify the original source code. I'll make the necessary changes in the PR.